julia_src/reopt_model.jl

```diff
@@ -1145,8 +1145,202 @@ function add_cost_function(m, p)
+		+ (p.STThermOutToThermInRatio-1) * sum(m[:dvThermalToSteamTurbine][tst,ts]*p.TechPercentRE[tst] for ts in p.TimeStep, tst in p.TechCanSupplySteamTurbine) )) # plus steam turbine generation from RE steam, minus RE steam feeding steam turbine
+		# TODO: subtract out steam turbine waste heat/curtailment and storage losses, add in GHP 
+	# Total heat - same equation without p.TechPercentRE factor
+	m[:AnnualHeatMMBTU] = @expression(m,p.TimeStepScaling*
+		(sum(p.ProductionFactor[t,ts] * p.LevelizationFactor[t] * m[:dvThermalProduction][t,ts] for t in p.HeatingTechs, ts in p.TimeStep) #total heat generation (need to see how GHP fits into this)
+		- sum(m[:dvProductionToWaste][t,ts] for t in p.CHPTechs, ts in p.TimeStep) #minus CHP waste heat
+		- sum(m[:dvProductionToStorage][b,t,ts]*(1-p.ChargeEfficiency[t,b]*p.DischargeEfficiency[b]) for t in p.HeatingTechs, b in p.HotTES, ts in p.TimeStep))) #minus thermal storage losses
+	end
+
+### Year 1 Emissions Calculations
+function add_yr1_emissions_calcs(m,p)
+	# Components:
+	yr1_emissions_from_fuelburningtechs_CO2, yr1_emissions_from_fuelburningtechs_NOx, yr1_emissions_from_fuelburningtechs_SO2, yr1_emissions_from_fuelburningtechs_PM25 = calc_yr1_emissions_from_onsite_fuel(m,p; tech_array=p.FuelBurningTechs)
+	yr1_emissions_from_boilertechs_CO2, yr1_emissions_from_boilertechs_NOx, yr1_emissions_from_boilertechs_SO2, yr1_emissions_from_boilertechs_PM25 = calc_yr1_emissions_from_onsite_fuel(m,p; tech_array=p.BoilerTechs)
+	m[:yr1_emissions_onsite_fuel_CO2] = yr1_emissions_from_fuelburningtechs_CO2 + yr1_emissions_from_boilertechs_CO2
+	m[:yr1_emissions_onsite_fuel_NOx] = yr1_emissions_from_fuelburningtechs_NOx + yr1_emissions_from_boilertechs_NOx
+	m[:yr1_emissions_onsite_fuel_SO2] = yr1_emissions_from_fuelburningtechs_SO2 + yr1_emissions_from_boilertechs_SO2
+	m[:yr1_emissions_onsite_fuel_PM25] = yr1_emissions_from_fuelburningtechs_PM25 + yr1_emissions_from_boilertechs_PM25
+
+	m[:yr1_emissions_from_elec_grid_CO2], m[:yr1_emissions_from_elec_grid_NOx], m[:yr1_emissions_from_elec_grid_SO2], m[:yr1_emissions_from_elec_grid_PM25]  = calc_yr1_emissions_from_elec_grid_purchase(m,p)
+	m[:yr1_emissions_offset_from_elec_exports_CO2], m[:yr1_emissions_offset_from_elec_exports_NOx], m[:yr1_emissions_offset_from_elec_exports_SO2], m[:yr1_emissions_offset_from_elec_exports_PM25] = calc_yr1_emissions_offset_from_elec_exports(m,p;tech_array=p.ElectricTechs)
+
+	# Total:
+	if p.IncludeExportedElecEmissionsInTotal
+		include_exported_elec_emissions_in_total = 1
+	else
+		include_exported_elec_emissions_in_total = 0
+	end
+	#"net" indicates net of what is getting counted, based on user selection of whether or not to include
+	m[:yr1_emissions_from_elec_grid_net_if_selected_CO2] = m[:yr1_emissions_from_elec_grid_CO2] - include_exported_elec_emissions_in_total*m[:yr1_emissions_offset_from_elec_exports_CO2]
```

Comment:
@hdunham correct- "net if user selected net, otherwise gross." Do you think we should remove "if_selected" or do you have another naming convention idea? We've gone back and forth on this one a bunch ("_total", "_overall", "_net", "_net_if_selected") and welcome your thoughts